tools/serverpod_cli/test/analyzer/entities/entity_analyzer/relation_keyword_test.dart

```diff
@@ -498,13 +507,160 @@ fields:
     var definition = SerializableEntityAnalyzer.extractEntityDefinition(
       protocol,
     );
+    SerializableEntityAnalyzer.resolveEntityDependencies([definition!]);
 
     SerializableEntityAnalyzer.validateYamlDefinition(
       protocol.yaml,
       protocol.yamlSourceUri.path,
       collector,
       definition,
-      [definition!],
+      [definition],
+    );
+
+    expect(
+      collector.errors.length,
+      greaterThan(0),
+      reason: 'Expected an error',
+    );
+
+    expect(
+      collector.errors.first.message,
+      'The "parent" property must be defined on id fields.',
+    );
+  });
+
+  test('Given a class ', () {
+    var collector = CodeGenerationCollector();
+
+    var protocol1 = ProtocolSource(
+      '''
+class: Example
+table: example
+fields:
+  parent: ExampleParent?, relation
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var protocol2 = ProtocolSource(
+      '''
+class: ExampleParent
+table: example_parent
+fields:
+  name: String
+''',
+      Uri(path: 'lib/src/protocol/example.yaml'),
+      [],
+    );
+
+    var definition1 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol1,
+    );
+
+    var definition2 = SerializableEntityAnalyzer.extractEntityDefinition(
+      protocol2,
+    );
+
+    var entities = [definition1!, definition2!];
+    SerializableEntityAnalyzer.resolveEntityDependencies(entities);
+
+    SerializableEntityAnalyzer.validateYamlDefinition(
+      protocol1.yaml,
+      protocol1.yamlSourceUri.path,
+      collector,
+      definition1,
+      entities,
+    );
+
+    var classDefinition = definition1 as ClassDefinition;
+
+    expect(
```

Comment:
Suggestion: Expect no errors to be collected.

We should probably expected no errors to be collected here as well right? My concern is that we otherwise measure success differently between this test and the following ones and both could be true at the same time.